libs/designsystem/tab-navigation/src/tab-navigation/tab-navigation.component.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  ViewChild,
+} from '@angular/core';
+import { filter, fromEvent, map, Subject, takeUntil, tap } from 'rxjs';
+import { WindowRef } from '@kirbydesign/designsystem/types';
+import { TabNavigationItemComponent } from '../tab-navigation-item/tab-navigation-item.component';
+
+const ARROW_LEFT = 'ArrowLeft';
+const ARROW_RIGHT = 'ArrowRight';
+
+@Component({
+  selector: 'kirby-tab-navigation',
+  templateUrl: './tab-navigation.component.html',
+  styleUrls: ['./tab-navigation.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TabNavigationComponent implements AfterViewInit, OnDestroy {
+  public readonly DEBOUNCE_TIME_MS = 250;
+
+  @ViewChild('tabBar')
+  private tabBar: ElementRef<HTMLElement>;
+
+  @ContentChildren(TabNavigationItemComponent, { read: ElementRef })
+  private tabs: QueryList<ElementRef<HTMLElement>>;
+
+  private tabBarElement: HTMLElement;
+  private tabElements = new Array<HTMLElement>();
+  private tabButtonElements = new Array<HTMLElement>();
+  private destroyed$ = new Subject<void>();
+
+  @Input()
+  get selectedIndex(): number {
+    return this._selectedIndex;
+  }
+
+  set selectedIndex(index: number) {
+    if (index !== this._selectedIndex) {
+      this._selectedIndex = index;
+
+      setTimeout(() => {
```

Comment:
The need for this timeout concerns me a bit, consider guarding for whatever makes this timeout necessary instead, and bail if e.g. the ContentChildren have not been set yet. In that case the similar code in AfterContentInit should do the necessary work if I read the code correctly.

Reply:
Well spotted :) - done!